src/Base/TimeElapsed.h

```diff
@@ -0,0 +1,100 @@
+/***************************************************************************
+ *   Copyright (c) 2011 Jürgen Riegel <juergen.riegel@web.de>              *
```

Comment:
Don't forget to add your copyright here.

Reply:
I always feel a bit ashamed to claim copyright on trivial refactors, but well... 
I would appreciate overall comment whenever those changes are making sense to you. I do not want to spend too much time going wrong direction :)
There's also FileInfo, which will be refactored next using std::filesystem.

Reply:
I love to see these changes -- FreeCAD's codebase is OLD, so much of it predates things like `recursive_mutex` and `filesystem`, it's great to get updated. As you are doing it just remember that we continue to support compilation on the oldest fully-supported Ubuntu LTS as our target "old" OS. That's what's holding us to C++17 right now, for example.

RE: Copyright... you be the judge, it's a balance between taking credit, and taking blame (e.g. don't assert that Jürgen wrote by himself a chunk of code that you've modified heavily. Of course git will record the truth, but I look at copyright headers a lot when deciding who to ask about a chunk of code).

Reply:
I'm always using git blame, but here it often fails due to lacking history. And yes, I know there is a tree with complete history imported.

C++17 is enough to get most of platform specific code gone. Just do not hold your breath, it is just a hobby project and I cannot spend much time on it. But the work will get eventually done - too much of my construction work now depends on FreeCAD, so I wish it will stay with me forever. That requires polishing here and there.

I'll probably convert this patchset to more general Base 'modernisation' branch and add more changes here, just to see how it all will work together.

Thank you for your time reviewing it.

Reply:
Thanks for the work you are doing. FYI you don't have to wait for it to be "perfect" or "finished" -- I'm more than happy to merge incremental improvements, it's much nicer than trying to review a PR that fixes everything all at once.

Reply:
I take care to make workflow bisectable, so it always works and compiles. That way it can be splitted into as many PR as feasible. Of course I'll do my best to make reviewing easier... (here I interrupt work as Addon Manager does not work with Qt6 - quite some deprecated and deleted functions still used. PR will be created).